janusgraph-test/src/test/java/org/janusgraph/core/ConfiguredGraphFactoryTest.java

```diff
@@ -0,0 +1,280 @@
+        final Map<String, Object> map = new HashMap<String, Object>();
+        map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+        map.put(GRAPH_NAME.toStringWithoutRoot(), "graph1");
+        configGraphManagement.createConfiguration(new MapConfiguration(map));
+        configGraphManagement.removeConfiguration("graph1");
+        thrown.expect(RuntimeException.class);
+        thrown.expectMessage(equalTo("Please create a template Configuration using the " +
+                                     "ConfigurationManagementGraph#createTemplateConfiguration API."));
+        ConfiguredGraphFactory.create("graph1");
+    }
+
+    @Test
+    public void updateConfigurationShouldOnlyUpdateForGraphAfterWeCloseAndReOpen() throws InterruptedException, BackendException, Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            map.put(GRAPH_NAME.toStringWithoutRoot(), "graph1");
+            configGraphManagement.createConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+            assertNotNull(graph);
+
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "bogusBackend");
+            configGraphManagement.updateConfiguration("graph1", new MapConfiguration(map));
+
+            final StandardJanusGraph graph1 = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+            assertNotNull(graph);
+
+            ConfiguredGraphFactory.close("graph1");
+
+            // we should throw an error since the config has been updated and we are attempting
+            // to open a bogus backend
+            thrown.expect(IllegalArgumentException.class);
+            thrown.expectMessage(equalTo("Could not find implementation class: bogusBackend"));
+            final StandardJanusGraph graph2 = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void shouldCreateTwoGraphsUsingSameTemplateConfiguration() throws InterruptedException, BackendException, Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            configGraphManagement.createTemplateConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph1 = (StandardJanusGraph) ConfiguredGraphFactory.create("graph1");
+            final StandardJanusGraph graph2 = (StandardJanusGraph) ConfiguredGraphFactory.create("graph2");
+
+            assertNotNull(graph1);
+            assertNotNull(graph2);
+
+            assertEquals(graph1.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
```

Comment:
Expected value (constant) is first argument to `assertEquals()`. Same for the next line as well.